packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -249,18 +240,34 @@ export class DataVisualizerGraph
 	private registerVisualizerForVisualizableObject(
 		visualizableObject: VisualizableFluidObject,
 	): FluidObjectId {
-		if (!this.visualizerNodes.has(visualizableObject.id)) {
+		// Store type check results to avoid recomputing
+		const isDataObj = isDataObject(visualizableObject);
+		const isTreeDataObj = isTreeDataObject(visualizableObject);
+
+		// Get the root shared object and its ID based on the object type
+		const rootSharedObject = isDataObj
+			? (visualizableObject as unknown as { readonly root: ISharedDirectory }).root
+			: isTreeDataObj
+				? (visualizableObject.sharedTree as unknown as ISharedObject)
+				: visualizableObject;
+
+		// Generate the fluidObjectId based on the object type
+		const fluidObjectId =
+			isDataObj || isTreeDataObj
+				? `${visualizableObject.id}-${rootSharedObject.id}`
```

Comment:
I've added wrapper functions `createDataObjectVisualizer()` & `createTreeDataObjectVisualizer()` to disntinguish the `fluidObjectId` of root `ISharedObject`. 

Since the `TreeDataObject.sharedTree` will always have a unique id, this maybe unnecessary for the `TreeDataObject`, but I've added this for consistency in pattern for non-`ISharedObject` data.